### PR TITLE
2021 Hyundai Elantra port

### PR DESCRIPTION
**Pull request by CruiseBrantley**
Fingerprint for 2021 Hyundai Elantra
Test Route: 82e9cdd3f43bf83e|2021-05-15--02-42-51

Minor issues left to resolve: 
- ~~Check Lane Follow Assist Warning~~ -Fixed
- Sometimes on brake message appears: "Smart Cruise Control Conditions Not Met"

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -577,10 +581,18 @@ class Buttons:
     (Ecu.fwdCamera, 0x7c4, None): [b'\xf1\x00JFA LKAS AT USA LHD 1.00 1.02 95895-D5000 h31'],
     (Ecu.transmission, 0x7e1, None): [b'\xf1\x816U2V8051\x00\x00\xf1\x006U2V0_C2\x00\x006U2V8051\x00\x00DJF0T16NL0\t\xd2GW'],
   },
+  CAR.ELANTRA_21: {
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/values.py:584]: add to the IGNORED_FINGERPRINTS list

**Comment by adeebshihadeh** [comment]
Are you sure this requires a new car? Why can't the existing ELANTRA be used?

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (CruiseBrantley) [reply selfdrive/car/hyundai/values.py:584]: Added.

**Comment by CruiseBrantley** [comment]
> Are you sure this requires a new car? Why can't the existing ELANTRA be used?

Model year is "All new" which also includes a new harness (A2 I believe) I used Sonata as a reference as another A harness vehicle, I assumed the features being different would be problematic.

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -21,6 +21,7 @@ def __init__(self, CP):
 class CAR:
   # Hyundai
   ELANTRA = "HYUNDAI ELANTRA 2017"
+  ELANTRA_21 = "HYUNDAI ELANTRA 2021"
```
Comment (pd0wm) [inline selfdrive/car/hyundai/values.py:24]:
```suggestion
  ELANTRA_2021 = "HYUNDAI ELANTRA 2021"
```

Let's stay consistent with the Ionic naming.

Comment (CruiseBrantley) [reply selfdrive/car/hyundai/values.py]: Got it, fixed.

**Comment by adeebshihadeh** [comment]
Thanks for the PR!

**Pull request merged**